tests/core/test_robust_cost.py

```diff
@@ -136,3 +136,53 @@ def test_mask_jacobians(loss_cls):
     torch.testing.assert_close(err, err_expected)
     for j1, j2 in zip(jac, jac_expected):
         torch.testing.assert_close(j1, j2)
+
+
+@pytest.mark.parametrize("batch_size", [1, 4])
+def test_flatten_dims(batch_size):
```

Comment:
Assuming we don't need any additional test for flatten_dims to verify compatibility with our internal vectorization?

Reply:
I think it should work fine unless one has a mix of similar cost functions but some have `flatten_dims=True` and others not. Then vectorization code will copy whichever flag value happens to find first, and apply it to all of them. And even in that case, I think the output value should be correct. 

However, this is actually a general issue that I found a while ago and I had forgotten about it. I'll open a Github bug report so I don't forget. But I'm not sure there is an easy work around for this. Basically, if your error depends on python variables not declared as optim or aux vars, and they are not shared across all cost functions, then vectorization might result in incorrect results.

Reply:
> error depends on python variables not declared as optim or aux vars
Hmm by definition shouldn't such variables be included in aux vars?

Reply:
If they are `theseus` variables then yes, this is natural. But this is kinda awkward for things like flags. Check out the example in #511. I could see this coming up even with neural nets in some corner cases.

Reply:
Ah okay makes sense. I wasn't thinking about flags.

Reply:
In the end I added a custom fix for `RobustCostFunction` so there shouldn't be any issues with vectorize either (see 648ba7842d413282f548e36abe9fc5d16d53bab4).

Reply:
Great, this is a clean fix and sufficient for now.